go/tools/gopackagesdriver/gopackagesdriver_test.go

```diff
@@ -89,6 +109,33 @@ const (
 	bzlmodOsPkgID = "@@io_bazel_rules_go//stdlib:os"
 )
 
+func TestStdlib(t *testing.T) {
+	resp := runForTest(t, DriverRequest{}, ".", "std")
+
+	if len(resp.Packages) == 0 {
+		t.Fatal("Expected stdlib packages")
+	}
+
+	for _, pkg := range resp.Packages {
+		// net, plugin and user stdlib packages seem to have compiled files only for Linux.
+		if pkg.Name != "cgo" {
+			continue
+		}
+
+		var hasCompiledFiles bool
+		for _, x := range pkg.CompiledGoFiles {
+			if filepath.Ext(x) == "" {
```

Comment:
Why is an empty extension the correct condition here?

Reply:
Maybe I didn't understand what you meant, but no, it's precisely the opposite. The test fails if the driver response returns an empty list of packages for the stdlib.

Reply:
> Why is an empty extension

Ok, I've just understood your question, sorry. This condition (empty file extension) is already used in `gopackagesdriver` to identify this kind of compiled file:

- [tools/gopackagesdriver/build_context.go#L26-L31](https://github.com/bazelbuild/rules_go/blob/master/go/tools/gopackagesdriver/build_context.go#L26-L31)

Reply:
Thanks for the explanation